lib/dor/assembly/content_metadata_from_stub/structural_builder.rb

```diff
@@ -34,11 +34,11 @@ def self.find_common_path(objects)
         end
         private_class_method :find_common_path
 
-        def self.administrative(file_attributes)
+        def self.administrative(file_attributes, object_access)
           {
             sdrPreserve: file_attributes[:preserve] == 'yes',
-            publish: file_attributes[:publish] == 'yes',
-            shelve: file_attributes[:shelve] == 'yes'
+            publish: object_access.view == 'dark' ? false : file_attributes[:publish] == 'yes',
+            shelve: object_access.view == 'dark' ? false : file_attributes[:shelve] == 'yes'
           }
```

Comment:
The logic beforehand deferred entirely to the stub content metadata when determining if a file should be set to publish or shelve. We now consider the object rights, and if dark, coerce to `false` for both.